opensrp-chw/src/main/java/org/smartregister/chw/sync/ChwSyncIntentService.java

```diff
@@ -1,9 +1,23 @@
 package org.smartregister.chw.sync;
 
+import org.smartregister.chw.dao.EventDao;
 import org.smartregister.sync.intent.SyncIntentService;
 
 public class ChwSyncIntentService extends SyncIntentService {
 
+    @Override
+    protected void handleSync() {
+        // fetch the last downloaded serverVersion before any unsyced data
+        Long serverVersion = EventDao.getMinimumVerifiedServerVersion();
+        if (serverVersion != null)
+            org.smartregister.util.Utils.getAllSharedPreferences().saveLastSyncDate(serverVersion);
+
+        // flag all contentious events as unsynced
+        EventDao.markEventsForReUpload();
```

Comment:
Why not add a test for this?

Reply:
@ekigamba  this is the Protected method writing a unit test might not be applicable for it